tools/crm_simulate.c

```diff
@@ -1139,9 +1139,15 @@ main(int argc, char **argv)
     }
 
   done:
-    pe_free_working_set(data_set);
-    global_cib->cmds->signoff(global_cib);
-    cib_delete(global_cib);
+    if (data_set) {
+        pe_free_working_set(data_set);
+    }
+
+    if (global_cib) {
+        global_cib->cmds->signoff(global_cib);
+        cib_delete(global_cib);
+    }
```

Comment:
The pointers obviously need guarding but the various free funcs (should) all check for NULL themselves

Reply:
Yeah, all those free functions are safe.  I could modify this block to just check when calling `global_cib->cmds->signoff`, or I could leave it as-is.